src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs

```diff
@@ -244,14 +258,48 @@ public static void InstallWebWorkerInterop(bool installJSSynchronizationContext,
 
         public static void UninstallWebWorkerInterop()
         {
-            var ctx = SynchronizationContext.Current as JSSynchronizationContext;
+            var ctx = JSSynchronizationContext.CurrentJSSynchronizationContext;
             var uninstallJSSynchronizationContext = ctx != null;
             if (uninstallJSSynchronizationContext)
             {
+                foreach (var jsObjectWeak in ThreadCsOwnedObjects.Values)
+                {
+                    if (jsObjectWeak.TryGetTarget(out var jso))
+                    {
+                        jso.Dispose();
+                    }
+                }
+                foreach (var gch in ThreadJsOwnedObjects.Values)
+                {
+                    GCHandle gcHandle = (GCHandle)gch;
+
+                    // if this is pending promise we reject it
+                    if (gcHandle.Target is TaskCallback holder)
+                    {
+                        unsafe
+                        {
+                            holder.Callback!.Invoke(null);
```

Comment:
Can this throw?

Reply:
I will change the implementation of the callback to use `TrySetException` instead of `SetException`. 
Should I rather make it `TrySetCanceled` ?

See `JSMarshalerArgument.Task.cs`